protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -93,43 +124,71 @@ fn execute_cheatcode_hint(
-                    .expect("path not resolved properly")
-                    .path()
-                    .to_str()
-                    .expect("failed to convert path to string")
-                    .to_string();
-                if path_str.contains(&contract_value_as_short_str[..])
-                    && path_str.contains(".sierra.json")
-                {
-                    maybe_sierra_path = Some(path_str);
+            let contract_value_as_short_str = as_cairo_short_string(&contract_value)
+                .expect("Converting contract name to short string failed");
+            let current_dir = std::env::current_dir()
+                .expect("Failed to get current directory")
+                .join("target/dev");
+
+            let mut paths = std::fs::read_dir(&current_dir)
+                .expect("Failed to read ./target/dev, maybe scarb build failed?");
+
+            let starknet_artifacts = &paths
+                .find_map(|path| match path {
+                    Ok(path) => {
+                        let name = path.file_name().into_string().ok()?;
+                        name.contains("starknet_artifacts").then_some(path)
+                    }
+                    Err(_) => None,
+                })
+                .expect("Failed to find starknet_artifacts.json file");
+            let starknet_artifacts = fs::read_to_string(starknet_artifacts.path())
+                .expect("Failed to read starknet_artifacts.json contents");
+            let starknet_artifacts: ScarbStarknetArtifacts =
+                serde_json::from_str(starknet_artifacts.as_str())
+                    .expect("Failed to parse starknet_artifacts.json contents");
+
+            let sierra_path = starknet_artifacts.contracts.iter().find_map(|contract| {
+                if contract.contract_name == contract_value_as_short_str {
+                    return Some(contract.artifacts.sierra.clone());
                 }
-            }
-            let file = std::fs::File::open(
-                maybe_sierra_path.expect("no valid path to sierra file detected"),
-            )
-            .expect("file should open read only");
-            let sierra_contract_class: ContractClass =
-                serde_json::from_reader(file).expect("file should be proper JSON");
+                None
+            }).unwrap_or_else(|| panic!("Failed to find contract {contract_value_as_short_str} in starknet_artifacts.json"));
+            let sierra_path = current_dir.join(sierra_path);
+
+            let file = std::fs::File::open(&sierra_path)
+                .unwrap_or_else(|_| panic!("Failed to open file at path = {:?}", &sierra_path));
+            let sierra_contract_class: ContractClass = serde_json::from_reader(&file)
+                .unwrap_or_else(|_| panic!("File to parse json from file = {file:?}"));
 
             let casm_contract_class =
                 CasmContractClass::from_contract_class(sierra_contract_class, true)
```

Comment:
Do we know why do we add pythonic hints do casm? Is there any reason for it?
cc @mkaput

Reply:
I think that's because you are running this code in pythonic VM?

Reply:
I don't think we are, are we?
cc @karol-bisztyga

Reply:
Where are we adding these?

Reply:
The `true` argument in function:
```rust
CasmContractClass::from_contract_class(sierra_contract_class, true)
```

Reply:
Changing it to false causes our declare tests to fail, so I guess it's needed.

Reply:
Can we add an issue to investigate it in the future? I guess we want to get rid of druciarstwo (at least partially)